File: backend/app/models/payment.rb
class Payment < ApplicationRecord
	belongs_to :host, class_name: "User", foreign_key: "host_id"
	belongs_to :customer, class_name: "User", foreign_key: "customer_id"

	validates_presence_of :price, :status, :host_id, :customer_id, :name
	validates :status, inclusion: { in: ['paid', 'ordered', 'cancelled'] }
end

File: flutter/lib/models/payment.dart
class Payment {
  final int id;
  final int hostId;
  final int customerId;
  final int price;
  final String status;
  final String name;

  Payment({
    required this.id,
    required this.hostId,
    required this.customerId,
    required this.price,
    required this.status,
    required this.name,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return new Payment(
      id: int.parse(json['id']),
      name: json['name'],
      price: int.parse(json['price']),
      status: json['status'],
      hostId: int.parse(json['hostId']),
      customerId: int.parse(json['customerId']),
    );
  }
}
